e3sm_warehouse/e3sm_warehouse.py

```diff
@@ -654,7 +655,7 @@ def add_args(
         )
         p.add_argument(
             "-w",
-            "--warehouse-path",
```

Comment:
I might need to revert these specific argument name changes since they probably don't make sense

Reply:
This is another example of term-overloading. We still refer to the user_pub filesystem location, to which archives datasets are extracted, as the "warehouse" (fully, /p/user_pub/e3sm/warehouse), and the "-w" of the command line allows a user to override that location.

Reply:
Note that there is a corresponding argument "-p", indicating the "publication" location of datasets that are published.  Often referred to as the "pub_dirs", that default is "/p/user_pub/work" an historical choice that I think is terrible, but necessary legacy access.

Reply:
Got it, I will revert these types of changes to `warehouse`. `e3sm_warehouse` will refer strictly to the Python package, not any paths.